core/heap.c

```diff
@@ -877,29 +877,30 @@ vmm_place_vmcode(vm_heap_t *vmh, size_t size, heap_error_code_t *error_code)
          * syslog or assert here
          */
         /* need extra size to ensure alignment */
-        vmh->alloc_size = size + DYNAMO_OPTION(vmm_block_size);
+        vmh->alloc_size = *size + DYNAMO_OPTION(vmm_block_size);
 #ifdef X64
         /* PR 215395, make sure allocation satisfies heap reachability contraints */
         vmh->alloc_start = os_heap_reserve_in_region(
             (void *)ALIGN_FORWARD(heap_allowable_region_start, PAGE_SIZE),
             (void *)ALIGN_BACKWARD(heap_allowable_region_end, PAGE_SIZE),
-            size + DYNAMO_OPTION(vmm_block_size), error_code, true /*+x*/);
+            *size + DYNAMO_OPTION(vmm_block_size), error_code, true /*+x*/);
 #else
         vmh->alloc_start = (heap_pc)os_heap_reserve(
-            NULL, size + DYNAMO_OPTION(vmm_block_size), error_code, true /*+x*/);
+            NULL, *size + DYNAMO_OPTION(vmm_block_size), error_code, true /*+x*/);
 #endif
         vmh->start_addr =
             (heap_pc)ALIGN_FORWARD(vmh->alloc_start, DYNAMO_OPTION(vmm_block_size));
         LOG(GLOBAL, LOG_HEAP, 1,
             "vmm_heap_unit_init unable to allocate at preferred=" PFX
             " letting OS place sz=%dM addr=" PFX "\n",
-            preferred, size / (1024 * 1024), vmh->start_addr);
+            preferred, *size / (1024 * 1024), vmh->start_addr);
         if (vmh->alloc_start == NULL && DYNAMO_OPTION(vm_allow_smaller)) {
             /* Just a little smaller might fit */
             size_t sub = (size_t)ALIGN_FORWARD(size / 16, 1024 * 1024);
             SYSLOG_INTERNAL_WARNING_ONCE("Full size vmm heap allocation failed");
-            if (size > sub)
-                size -= sub;
+            /* Don't go too small. */
+            if (*size > 2 * sub)
```

Comment:
Should this be >= or does it really need to be greater?  Assuming these are all aligned sizes that could be a meaningful difference.

Reply:
I think the original >sub or written as >=2*sub are equivalent; I was trying to keep it slightly larger and not drop all the way to 1M as happened in the bug but 2M isn't much better so maybe it's not a worthwhile heuristic change.

Reply:
To answer your question more directly it could be = and not go to 0 or something catastrophic; I was trying to keep it a little bit bigger.  Leaning toward just leaving it this way.